### PR TITLE
ROX-13246: add Clair v4 Image Integration

### PR DESCRIPTION
**Pull request by RTann**
## Description

This is currently unfinished and behind a feature flag. This is just the start to the implementation to add Clair v4 support to make reviews more manageable.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

Not at the moment, as this is behind a feature flag and will be completed with followup(s)

### CODE REVIEW

File: pkg/scanners/clairv4/clairv4.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/httputil/proxy"
+	"github.com/stackrox/rox/pkg/registries"
+	"github.com/stackrox/rox/pkg/scanners/types"
+	"github.com/stackrox/rox/pkg/urlfmt"
+)
+
+const (
+	requestTimeout = 2 * time.Minute
```
Comment (md2119) [inline pkg/scanners/clairv4/clairv4.go:19]: Seems high for a timeout.

File: pkg/scanners/clairv4/clairv4.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	indexReportEndpoint         string
+	indexEndpoint               string
+	vulnerabilityReportEndpoint string
+}
+
+func newScanner(integration *storage.ImageIntegration, activeRegistries registries.Set) (*clairv4, error) {
+	cfg := integration.GetClairV4()
+	if cfg == nil {
+		return nil, errors.New("Clair v4 configuration required")
+	}
```
Comment (md2119) [inline pkg/scanners/clairv4/clairv4.go:55]: You can move this into `validate()`

**Review by md2119 — APPROVED** [review]
The blueprint LGTM

File: pkg/scanners/clairv4/clairv4.go (hunk 1 above)
Comment (RTann) [reply pkg/scanners/clairv4/clairv4.go:19]: This was required when I tested it locally [here](https://github.com/stackrox/stackrox/pull/3768). Going to keep it for now, but we can always update if needed

**Pull request merged**